src/cider/nrepl.clj

```diff
@@ -37,12 +37,11 @@
                   (-> min-clojure-verion :major long))
                (< (-> *clojure-version* :minor long)
                   (-> min-clojure-verion :minor long))))
-  (try
-    (.println System/err (format "cider-nrepl requires a newer Clojure version (found: %s, minimum required: %s). Exiting."
-                                 *clojure-version*
-                                 min-clojure-verion))
-    (finally
-      (System/exit 1))))
+  (let [msg (format "cider-nrepl requires a newer Clojure version (found: %s, minimum required: %s)."
+                    *clojure-version* min-clojure-verion)]
+    (try
+      (.println System/err msg)
+      (finally (throw (ex-info msg {}))))))
```

Comment:
I don't like that a library can System/exit, I think it takes too much open itself. Throwing an exception is better because it can be caught (for whichever reason the user has), it can be properly logged, you get a stacktrace to know who required `cider.nrepl` namespace (e.g. this might be a surprise to the user), etc.

Reply:
Fine by me - that `System/exit` was always controversial. Ideally we'd have at some point a way for the server to notify clients if they are not meeting some runtime requirements, but if cider-nrepl can work to some extent and the users are aware of this, it's probably a good trade-off.

Reply:
Just to clarify, if user doesn't try to catch the exception around requiring `cider.nrepl`, then the end result will be the same as in System/exit – the server will fail to start. It's just that throwing an exception gives more options to react to it and properly report.

Reply:
Yeah, yeah - I get this. My point was mainly that the way the code was previously written it would fail on the version check alone, even if it might actually work in practice and might be fine with some users. (working, but not really supported) I think the purpose of the explicit `exit` here was to prevent such scenarios from happening. 

I agree it's better for the clients to decide how to handle an error from a library that they're using.